.github/workflows/porter.yml

```diff
@@ -0,0 +1,172 @@
+name: porter/porter
+on:
+  pull_request:
+    branches:
+    - main
+env:
+  GOCACHE: "${{ runner.workspace }}/.cache/go-build/"
+  GOCACHE_KEY: v3 | go-build | "${{ runner.os }}" | go.sum
+  GOCACHE_RESTOREKEYS: |
+    v3 | go-build | "${{ runner.os }}"
+    v3 | go-build | "${{ runner.os }}" | go.sum
+  GOMODCACHE: "/home/vsts/go/pkg/mod"
+  GOMODCACHE_KEY: v4 | go-pkg | "${{ runner.os }}" | go.sum
+  GOMODCACHE_RESTOREKEYS: |
+    v4 | go-pkg | "${{ runner.os }}"
+    v4 | go-pkg | "${{ runner.os }}" | go.sum
+  GOVERSION: 1.20.7
+  GOXCACHE_KEY: v3 | go-xbuild | "${{ runner.os }}" | go.sum
+  GOXCACHE_RESTOREKEYS: |
+    v3 | go-xbuild | "${{ runner.os }}"
+    v3 | go-xbuild | "${{ runner.os }}" | go.sum
+jobs:
+  Setup-pr_type:
+    name: Determine PR Type
+    runs-on: ubuntu-latest
+    steps:
+    - name: checkout
+      uses: actions/checkout@v3.5.0
+    - name: Is Doc Only?
+      id: BUILD
+      run: "./build/doc-only-build.sh"
+      shell: bash
+  Validate-build:
+    name: Native Compile
+    needs:
+    - Setup-pr_type
+    runs-on: ubuntu-latest
+    if: success() && needs.Setup-pr_type.outputs['pr_type.BUILD.DOCS_ONLY'] == 'false'
+    steps:
+    - name: checkout
```

Comment:
Do we need to checkout twice?  Can we checkout once and see if it is a doc build, etc?